calico/felix/fetcd.py

```diff
@@ -917,18 +979,44 @@ def parse_endpoint(config, combined_id, raw_json):
     return endpoint
 
 
-def parse_rules(profile_id, raw_json):
+def parse_tier_data(tier, data):
+    data = safe_decode_json(data, log_tag="tier %s" % tier)
+    try:
+        common.validate_tier_data(tier, data)
+    except ValidationFailed as e:
+        _log.error("Validation failed for tier data for tier %s: %r",
+                   tier, e)
+        return None
+    else:
+        return data
+
+
+def parse_profile(profile_id, raw_json, require_selector=False,
```

Comment:
Isn't require_selector always False, for profiles?  Anyway it seems unused, so would be nice to simplify this and callers.